### ISSUE TITLE
TLS not working as expected (?) 

### ISSUE THREAD

**Issue by mreferre** [vmware/vic#2571]
@mlh78750 @corrieb 

I have been talking to @stuclem about this but we couldn’t figure out what’s going on. 

In a nutshell, our understanding is that if you deploy a VCH by default it is TLS enabled and you need to use the certificates generated to be able to connect to it. 

This is the string I am using to deploy mine: 

       ./vic/vic-machine-darwin create --name VCH1 \
            --target "${VCENTERSERVER}" \
            --user "${VSPHEREUSER}" \
            --password "${VSPHEREPASSWORD}" \
            --bridge-network "${VCHINTERNALNETWORK}" \
            --external-network "${EXTERNALNETWORK}" \
            --image-store "${VSPHEREDATASTORE}"  \
            --docker-insecure-registry "${HARBORVMIP}”

The weird thing about this is that I can connect to this deamon from literally everywhere I want (different laptops etc) and still get access to it. My VCH has IP `192.168.1.195` and here is what I can do from a laptop that I have never used before (where I have just installed Docker and never downloaded vic-machine ever):
```
mreferre-m01:~ mreferre$ export DOCKER_HOST=192.168.1.195:2376

mreferre-m01:~ mreferre$ docker ps
Get http://192.168.1.195:2376/v1.24/containers/json: malformed HTTP response "\x15\x03\x01\x00\x02\x02".
* Are you trying to connect to a TLS-enabled daemon without TLS?

mreferre-m01:~ mreferre$ docker --tls ps
Error response from daemon: client is newer than server (client API version: 1.24, server API version: 1.23)

mreferre-m01:~ mreferre$ export DOCKER_API_VERSION=1.23

mreferre-m01:~ mreferre$ docker --tls ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES

mreferre-m01:~ mreferre$ docker --tls images
REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE

mreferre-m01:~ mreferre$ docker --tls pull alpine
Using default tag: latest
Pulling from library/alpine
c0cb142e4345: Pull complete 
Digest: sha256:d019def466a5d56146bc49f16eeab517d3e007b60a1811bcf899b624cfb98cd6
Status: Downloaded newer image for library/alpine:latest

mreferre-m01:~ mreferre$ docker --tls run -it alpine /bin/ash
/ # 
```

I have never copied the certificates generated by vic-machine on this laptop. The first docker thing that this laptop has ever done is what you see above. And yet it was able to instantiate a ContainerVM on the VCH. 

We thought this was the behaviour if you use the `--no-tls` flag in vic-machine at VCH deployment time (that would allow you to connect un-authenticated to the VCH on port `2375`). Right now it seems I am able to connect un-authenticated to the VCH on port `2376`. 

Thoughts?

**Comment by stuclem** [comment]
Incidentally, I attempted the same exercise yesterday: deployed a VCH by running `vic-machine` in a Windows Server 2008 VM. Attempted to connect a Docker client running on my Windows 7 laptop to the VCH, without having copied over the certificates, and got the following error:

![cert](https://cloud.githubusercontent.com/assets/16718369/19115109/fe88bf76-8b10-11e6-88af-5ce74f0d6470.png)

So, things appear to be working as the they should do for me, inasmuch as I can't connect without copying the certs to the Docker client machine.

**Comment by mreferre** [comment]
Yes, it does sound something local (or an edge case I ended up in) but I am not sure what it is. The fact that I could connect (to 2376, with not certificates) from a laptop that has basically pretty much never been powered on is driving me nuts.

**Comment by mlh78750** [comment]
it appears there was a commit that turned off verification. This is a bug.

**Comment by mdubya66** [comment]
what build was this in?

**Comment by mdubya66** [comment]
@hickeng is this the culprit? https://github.com/vmware/vic/blob/master/lib/install/management/appliance.go#L651

**Comment by dougm** [comment]
Nothing has changed here guys, that's how `--tls` without `--tlsverify` is expected to behave and we've been using it that way all along.  Support for `--tlsverify` and friends is on its way in #2553

Related #1085

Closing as a dup of #2197

**Comment by anchal-agrawal** [comment]
@mdubya66 I believe that is unrelated; that change applies only to the client that makes a `docker info` request during installation to check that the appliance is ready.

**Comment by dougm** [comment]
@stuclem please post the output of the 'SET' command so we can see your shell's environment.  Were you using docker-machine / docker-machine env too?

**Comment by stuclem** [comment]
@dougm I am using Docker for Windows on both my laptop and in the VM. Nowadays I do all Docker and `vic-machine` operations in the VM. But unlike @mreferre and his virgin laptop, I have done other Docker stuff on my laptop ages ago, and there was some old Docker config knocking about:

![docker](https://cloud.githubusercontent.com/assets/16718369/19151612/e68c196e-8bcd-11e6-9d60-22a4538fdd26.png)

So, this is why I got the failure message above. The `DOCKER_HOST` variable is pointing to an ancient VCH that I deleted ages ago.
